src/huggingface_hub/snapshot_download.py

```diff
@@ -9,8 +9,9 @@
 from .utils import logging
 
 
-REPO_ID_SEPARATOR = "__"
-# ^ make sure this substring is not allowed in repo_ids on hf.co
+REPO_ID_SEPARATOR = "--"
+# ^ this substring is not allowed in repo_ids on hf.co
+# and is the canonical one we use for serialization of repo ids elsewhere.
```

Comment:
(means that you could get the repo id from its snapshot path on disk, if ever needs be)